lib/resty/websocket/client.lua

```diff
@@ -161,6 +168,15 @@ function _M.connect(self, uri, opts)
         return 1
     end
 
+    local custom_headers
+    if headers then
+      local tmp = {}
+      for k, v in pairs(headers) do
+        tmp[#tmp + 1] = "\r\n" .. k .. ": " .. tostring(v)
+      end
+      custom_headers = table.concat(tmp)
```

Comment:
We should avoid one table lookup of the `concat` field in the `table` table on such a hot code path.

Reply:
Fixed.